operators/infinispan/2.2.1/manifests/infinispan-operator.clusterserviceversion.yaml

```diff
@@ -0,0 +1,629 @@
+          - caches/finalizers
+          - caches/status
+          verbs:
+          - create
+          - get
+          - list
+          - patch
+          - update
+          - watch
+        - apiGroups:
+          - infinispan.org
+          resources:
+          - infinispans
+          - infinispans/finalizers
+          - infinispans/status
+          verbs:
+          - create
+          - get
+          - list
+          - patch
+          - update
+          - watch
+        - apiGroups:
+          - infinispan.org
+          resources:
+          - restores
+          - restores/finalizers
+          - restores/status
+          verbs:
+          - create
+          - get
+          - list
+          - patch
+          - update
+          - watch
+        - apiGroups:
+          - integreatly.org
+          resources:
+          - grafanadashboards
+          verbs:
+          - create
+          - delete
+          - get
+          - list
+          - update
+          - watch
+        - apiGroups:
+          - monitoring.coreos.com
+          resources:
+          - servicemonitors
+          verbs:
+          - create
+          - delete
+          - get
+          - list
+          - update
+          - watch
+        - apiGroups:
+          - networking.k8s.io
+          resources:
+          - customresourcedefinitions
+          - customresourcedefinitions/status
+          verbs:
+          - get
+          - list
+        - apiGroups:
+          - networking.k8s.io
+          resources:
+          - ingresses
+          verbs:
+          - create
+          - delete
+          - deletecollection
+          - get
+          - list
+          - update
+          - watch
+        - apiGroups:
+          - route.openshift.io
+          resources:
+          - routes
+          - routes/custom-host
+          verbs:
+          - create
+          - delete
+          - deletecollection
+          - get
+          - list
+          - update
+          - watch
+        serviceAccountName: infinispan-operator
+    strategy: deployment
+  installModes:
+  - supported: true
+    type: OwnNamespace
+  - supported: true
+    type: SingleNamespace
+  - supported: true
+    type: MultiNamespace
+  - supported: true
+    type: AllNamespaces
+  keywords:
+  - infinispan
+  - key value
+  - database
+  - datagrid
+  - open source
+  links:
+  - name: Blog
+    url: https://blog.infinispan.org/
+  - name: Documentation
+    url: https://infinispan.org/
+  - name: Chat
+    url: https://infinispan.zulipchat.com/#narrow/stream/185835-infinispan-cloud
+  - name: Quickstart
+    url: https://github.com/infinispan/infinispan-simple-tutorials/tree/main/operator
+  - name: Operator Source Code
+    url: https://github.com/infinispan/infinispan-operator
+  maintainers:
+  - email: infinispan-dev@lists.jboss.org
+    name: Infinispan Community
+  maturity: alpha
+  provider:
+    name: Infinispan
+  replaces: infinispan-operator.v2.1.5
```

Comment:
@ryanemerson I think you want to replace `infinispan-operator.v2.2.0`, since this version is in same channel `2.2.x`

Reply:
@mvalarh The intention was to skip the `2.2.0` release, `olm.skipRange: '>= 2.2.0 <2.2.1'`., so I thought it was necessary to have `replaces: infinispan-operator.v2.1.5`. Should this still be `replaces: infinispan-operator.v2.2.0`?

Reply:
@mvalarh @J0zi Can you clarify this?

I have tested locally, and if I use a combination of `replaces: infinispan-operator.v2.2.0` and `olm.skipRange: '>= 2.2.0 <2.2.1' then the upgrade path is `2.1.5 -> 2.2.0 -> 2.2.1`, however if I have ``replaces: infinispan-operator.v2.1.5` then the upgrade path is `2.15 -> 2.2.1` which is what we desire.

Reply:
@ryanemerson cross channel replaces does not work. Keep replaces inside the channel or add version you want to the affected channel.

Reply:
@J0zi What CSV configuration do I need to allow the first release in a channel to be skipped when changing channel.

i.e. 
Channel 2.1.x: v2.1.5
Channel 2.2.x: v2.2.1

What we desire, is that when the channel is manually changed from 2.1.x to 2.2.x the upgrade path is `2.1.5 -> 2.2.1`, not `2.1.5 -> 2.2.0 -> 2.2.1` ... or is this not possible?

Reply:
> What we desire, is that when the channel is manually changed from 2.1.x to 2.2.x the upgrade path is `2.1.5 -> 2.2.1`, not `2.1.5 -> 2.2.0 -> 2.2.1` ... or is this not possible?

BTW if this is not possible somehow, this seems like a serious limitation. If the `v2.2.0` release contained a CVE, then it would mean that the user has to install a compromised release in order to get from `2.1.5 -> 2.2.1`

Reply:
> @ryanemerson cross channel replaces does not work. Keep replaces inside the channel or add version you want to the affected channel.

I'm probably misunderstanding, but isn't that exactly what the released `v2.2.0` did?

https://github.com/k8s-operatorhub/community-operators/blob/main/operators/infinispan/2.2.0/manifests/infinispan-operator.clusterserviceversion.yaml#L449

Reply:
@mvalarh If we just remove the version, what's the upgrade path for users who have already installed `v2.2.0` from the CatalogSource? They still need to be able to transition from `v2.2.0 -> v2.2.1` when it's released.

Reply:
@J0zi @mvalarh Can you provide clarify on this or point me in the direction of somebody who can? We've wanted to get this release out for almost two weeks at this stage.

Reply:
> @ryanemerson cross channel replaces does not work. Keep replaces inside the channel or add version you want to the affected channel.

@ryanemerson as I wrote here ^, you cannot replace something, which is not in the channel. So, you can add `v.2.2.1` to  `2.1.x` channel. But it does not make sense for me. On the other hand, it will work. 
Every time you replace something, it should exist in the channel you are in. You can consider soem channel containing all operators for your case, let's say `stable`.

Reply:
Thank you @ryanemerson for explaining it in details. Now I get it. I think, that removing replaces which is pointing to another channel could help. But please give me a time to double check.